### PR TITLE
Feature/24 update flight api

### PR DESCRIPTION
**Pull request by dilyara4949**
Add api that updates flight info

### CODE REVIEW

File: internal/service/flight.go [hunk 1]
```diff
@@ -42,3 +43,33 @@ func (service *Flight) Get(ctx context.Context, id uuid.UUID, available bool) (*
...
+	if req.Destination != "" {
+		flight.Destination = req.Destination
+	}
+	if req.Rank != "" {
+		flight.Rank = req.Rank
+	}
+	if req.Price != 0 {
+		flight.Price = req.Price
+	}
+	if req.TotalTickets != 0 {
```
Comment (krylphi) [inline internal/service/flight.go:71]: You might still should to be able to set price and total tickets to 0. But since we want to check whether it's "real zero" or default value, you should consider using pointers to int. Here's a [bit more about this usage](https://www.reddit.com/r/golang/comments/qdgc1q/pointers_in_structures_for_json/). You also would need to revise `Validate()` method and pay attention regarding dereferencing.

File: internal/database/postgres/migration/000004_alter_users_add_not_null_to_role.down.sql
Comment (krylphi) [inline internal/database/postgres/migration/000004_alter_users_add_not_null_to_role.down.sql:1]: Why is this file renamed? Migration file name should remain unchanged, AFAIK, because it's one of the criteria for migration to pick files up for execution.

Comment (dilyara4949) [reply internal/database/postgres/migration/000004_alter_users_add_not_null_to_role.down.sql:1]: since i created new migration same time at different branches, they had same version, which will  cause error while migrating

Comment (krylphi) [reply internal/database/postgres/migration/000004_alter_users_add_not_null_to_role.down.sql:1]: I would suggest using an alternative versioning in that case. timestamps perhaps. this way collisions should not occur

**Review by krylphi — APPROVED** [review]
